### PR TITLE
Return to the same place when triggering a DAG

### LINKED ISSUE #20197: Triggering a DAG in Graph view switches to Tree view
### Apache Airflow version

2.2.2 (latest released)

### What happened

After I trigger a DAG from the Graph view, the view switches to the Tree view (happens on both 2.2.2 and main).

### What you expected to happen

I expect to stay in the Graph view, ideally switch to the newly created DAG run

### How to reproduce

Trigger a DAG from the Graph view

### Operating System

N/A

### Versions of Apache Airflow Providers

N/A

### Deployment

Astronomer

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by norm**
Rather than always going to the DAG's default view, return to the same view as you were on.

closes: #20197

### CODE REVIEW

**Review by bbovenzi — APPROVED** [review]
Nice work!

File: airflow/www/templates/airflow/dag.html [hunk 1]
```diff
@@ -153,11 +153,11 @@ <h4 class="pull-right" style="user-select: none;-moz-user-select: auto;">
                   <input type="hidden" name="csrf_token" value="{{ csrf_token() }}">
                   <input type="hidden" name="dag_id" value="{{ dag.dag_id }}">
                   <input type="hidden" name="unpause" value="True">
-                  <input type="hidden" name="origin" value="{{ url_for('Airflow.' + dag.default_view, dag_id=dag.dag_id) }}">
+                  <input type="hidden" name="origin" value="{{ request.full_path }}">
```
Comment (ashb) [inline airflow/www/templates/airflow/dag.html]: Argh, full_path includes the  query string, which is a bit of a problem.

If you have changed the date controls and are viewing an "old" run (in time, not out of date) then the run you have just triggered won't be visible. Not great UX.

Comment (norm) [reply airflow/www/templates/airflow/dag.html]: Agreed, I'll rework it.

**Review by ashb — APPROVED** [review]

File: tests/www/views/test_views_tasks.py [hunk 2]
```diff
@@ -272,14 +272,30 @@ def test_dag_details_trigger_origin_tree_view(app, admin_client):
...
+    )
+
+    url = 'dag_details?dag_id=test_tree_view'
+    resp = admin_client.get(url, follow_redirects=True)
+    params = {'dag_id': 'test_tree_view', 'origin': '/dag_details?dag_id=test_tree_view'}
+    href = f"/trigger?{html.escape(urllib.parse.urlencode(params))}"
+    check_content_in_response(href, resp)
+
+
+def test_dag_details_trigger_origin_dag_details_view(app, admin_client):
```
Comment (ashb) [inline tests/www/views/test_views_tasks.py:298]: Same function name makes flake8 a sad bunny

Comment (uranusjr) [reply tests/www/views/test_views_tasks.py:298]: Not just flake8, this would actually make the first test not run.

Comment (norm) [reply tests/www/views/test_views_tasks.py:298]: I am a copy and paste genius. Changed the name and made the intent of the seeming duplicate test a little clearer, hopefully.

**Review by uranusjr — APPROVED** [review]

**Comment by jedcunningham** [comment]
Kicking this to 2.2.5 because of a bug that was discovered.

**Pull request merged**